### ISSUE TITLE
co-maintain the package in conda-eda?

### ISSUE THREAD

**Issue by proppy** [flaport/condalayout#1]
Thanks for working on this package!

We're maintaining a set of eda related packages at https://github.com/hdl/conda-eda and were also looking at packaging klayout (https://github.com/hdl/conda-eda/issues/175).

What would you think of co-maintaining this package in the conda-eda repository? That could make it easier for users to install all eda related from one channel and allow all project to share the same continuous integration infrastructure for rebuilding and publishing packages.

Let me know what you think!

/cc @joamatab @mithro

**Comment by flaport** [comment]
Hey @proppy  that sounds good to me! 

But as you can maybe see this project is still pretty experimental. I think I'll keep on working in this repository until the build recipe has stabilized a bit.

Once I'm happy with the recipe (GUI + python package, windows & linux), I'd be happy to contribute it to conda-eda!

**Comment by proppy** [comment]
another thing to consider could be to contribute the package to conda-forge.

I recently went thru the staging process for magic in https://github.com/conda-forge/staged-recipes, so I'm happy to help navigating it if needed.

**Comment by joamatab** [comment]
How could we make a conda package for gdsfactory as well?

**Comment by flaport** [comment]
@proppy , yes, I'd love to add it to conda-forge. I once added my [fdtd](https://github.com/flaport/fdtd) package to conda-forge, so I'm pretty aware on how to do it myself too. The main problem is still that I cannot seem to build the GUI for any other version than python 3.8 for now... That said, publishing the klayout python package (not the GUI) on the other hand is basically just a repackaging of the PyPI wheel, so that should not be too difficult (I hope). Maybe we should start with that.

@joamatab , you can just repackage your gdsfactory wheels as conda packages quite easily. You just have to make sure you don't have any hard dependencies (i.e. ImportError if not present) on non-conda packages (like klayout for now). I can walk you through it in one of our next meetings if you want.

**Comment by proppy** [comment]
> The main problem is still that I cannot seem to build the GUI for any other version than python 3.8 for now.

Do you have some pointers on the issues you encountered? conda-forge builder seem to have pretty good support for building packages for multiple versions for python: https://anaconda.org/conda-forge/opencv/files

**Comment by saicharan0112** [comment]
@flaport I tried installing the conda package but it removes openroad and yosys packages.. To reproduce this issue, try this - 
```
bash> docker run -it conda/miniconda3
docker> conda install -c litex-hub yosys openroad open_pdks.sky130a magic netgen -y
docker> conda config --append channels conda-forge
docker> conda install -c flaport klayout -y
```
You would be seeing openroad and yosys under the removed packages list when you install the klayout package
![image](https://user-images.githubusercontent.com/100982953/181893569-d9486f75-64d9-4b3a-b1d3-f66ae54006c9.png)

**Comment by flaport** [comment]
Hi @saicharan0112 ,

It seems that those two packages only exist for python 3.7

Currently my klayout builds only exist for python 3.8

Hence installing klayout will upgrade python and remove those two packages.

**Comment by saicharan0112** [comment]
@flaport Oh ok. So what could be the possible way if I want to keep both of them and install klayout?

**Comment by joamatab** [comment]
You can pip install klayout 

Or sudo apt install klayout

**Comment by saicharan0112** [comment]
Yeah @joamatab , but the apt package manager version is at least 10 versions old. Current Klayout is 0.28 (github) but the package manager is still behind (0.26_2), if I am not wrong (https://packages.ubuntu.com/focal/klayout).. I am looking more into building the environment only with conda packages on a miniconda3 docker image. 
However, I managed to go with the deb file from the website (Thanks to @proppy for the suggestion) on an ubuntu image but it would be good to have a conda package compatibility here. 

https://github.com/saicharan0112/OpenFASOC/blob/main/docker/Dockerfile

**Comment by saicharan0112** [comment]
I don't know if it is a good idea to consider downgrading the current klayout package to python3.7 (or upgrading others to python3.8) just to have all tools running in a same environment since packages are really making things easy when it comes to installing tools in no time.

**Comment by proppy** [comment]
@flaport are you still open to co-maintain this package in conda-eda and/or conda-forge, there we could ensure it has a coherent environment with other related EDA packages?

**Comment by flaport** [comment]
Hi @proppy , I'd love to hand over the project such that someone else can maintain it actually. Unfortunately I only got it to build for python 3.8 and I'm pretty much stuck for the other python versions. Feel free to give it a shot yourself ;-)

**Comment by msaligane** [comment]
@proppy @saicharan0112 any update on this?

**Comment by proppy** [comment]
@msaligane let's discuss in https://github.com/hdl/conda-eda/issues/175 what it would take to maintain it in conda-eda.

**Comment by saicharan0112** [comment]
@flaport Can I know where you found these links?https://github.com/flaport/condalayout/blob/master/klayout-pypi-links.txt
By any chance, do you know where to find the pypi link for the 3.7 version?

**Comment by flaport** [comment]
If I remember correctly I just went to pypi.org and right clicked 'copy url' the download links

**Comment by saicharan0112** [comment]
@flaport Got it! And how did you manage to find the hash/link for different python versions for the same klayout version?

**Comment by saicharan0112** [comment]
Got it.. Its in the same location but need to scroll down.
